include/seqan3/core/algorithm/all.hpp

```diff
@@ -38,13 +38,68 @@
+ * This module provides a basic system to manage the configurations of algorithms using a unified interface.
+ *
+ * ### Usage
+ *
+ * The basis of any algorithm configuration are configuration elements. These are objects that handle a specific
+ * setting and must satisfy the seqan3::detail::config_element_concept. The following snippet demonstrates a
+ * basic setup for such configuration elements.
+ *
+ * \snippet test/snippet/core/algorithm/configuration.cpp configuration_setup
+ *
+ * Here, two classes with the name `bar` and `foo` are created. They can be normal or template classes and must
+ * inherit from seqan3::pipeable_config_element and contain a member with the name `value` to satisfy the respective
+ * concept. The separate `value` member is used for a proper encapsulation from the actual setting parameter.
+ * For example the alignment algorithms require a scoring scheme, but the scoring scheme itself should not be pipeable
+ * with other settings.
+ *
+ * In addition an enum type was defined that will be used later to allow for compatibility checks when combining
+ * settings in a configuration object. This enum assigns every configuration element a unique id.
+ * To provide compatibility checks the `seqan3::detail::compatibility_table` must be overloaded for the specific
+ * algorithm configuration.
+ *
+ * \snippet test/snippet/core/algorithm/configuration.cpp compatibility
+ *
+ * The type for the configuration element ids is used to overload the bool table. In the example above, both
+ * elements can be combined with each other but not with themselves, to avoid inconsistent settings.
+ *
+ * ### Combining Configurations
+ *
+ * To enable easy creation of algorithm settings the seqan3::configuration supports a pipeable interface for the
+ * different configuration elements which is added through the seqan3::pipeable_config_element base class.
+ * The following snippet demonstrates how `bar` and `foo` can be combined.
+ *
+ * \snippet test/snippet/core/algorithm/configuration.cpp combine
+
+ * ### Access the data
+ *
+ * The configuration inherits from a std::tuple and exposes a tuple like interface using the standard
+ * position-based and type-based `get` interfaces. The `get` interface was extended to also support
+ * template template types as input template parameters to query the correct element:
+ *
+ * \snippet test/snippet/core/algorithm/configuration.cpp get
+ *
+ * The get interface returns a reference to the stored configuration element. In some cases, e.g. the implementor
+ * of the actual algorithm, one wants to have an easy access to the actual value of the setting. Since, the
+ * configuration must not contain all possible configuration elements the seqan3::configuration provides a
+ * seqan3::configuration::value_or interface, which provides direct access to the value of the respective
+ * configuration element or uses a default value if the queried type is not contained in the configuration.
```

Comment:
> The get interface returns a reference to the stored configuration element. In some cases, e.g. the implementor  of the actual algorithm, one wants to have an easy access to the actual value of the setting. Since, the
  configuration must not contain all possible configuration elements the seqan3::configuration provides a
  seqan3::configuration::value_or interface, which provides direct access to the value of the respective
 configuration element or uses a default value if the queried type is not contained in the configuration.

Maybe shorter: 
The get interface returns a reference to the stored configuration element. In order to avoid an error if an optional confugartion element was not given, you can use the seqan3::configuration::value_or interface to give a default value:

Reply:
Well, except for bugs in the documentation I would prefer not to touch this stuff a lot. My plan is for the upcoming sprints to refactor the documentation as well, when the whole systems starts to grow together.

Reply:
ok